src/test/java/net/minecraftforge/debug/world/MultiNoiseBiomeSourceLoadingEventTest.java

```diff
@@ -0,0 +1,259 @@
+import net.minecraft.world.level.dimension.LevelStem;
+import net.minecraft.world.level.levelgen.NoiseBasedChunkGenerator;
+import net.minecraft.world.level.levelgen.NoiseGeneratorSettings;
+import net.minecraft.world.level.levelgen.synth.NormalNoise.NoiseParameters;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.RegisterCommandsEvent;
+import net.minecraftforge.event.world.MultiNoiseBiomeSourceLoadingEvent;
+import net.minecraftforge.eventbus.api.IEventBus;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.forge.event.lifecycle.GatherDataEvent;
+
+@Mod(MultiNoiseBiomeSourceLoadingEventTest.MODID)
+public class MultiNoiseBiomeSourceLoadingEventTest
+{
+    public static final String MODID = "multi_noise_biome_source_loading_event_test";
+    public static final Logger LOGGER = LogManager.getLogger();
+    
+    public static final ResourceLocation PRESET_TEST = new ResourceLocation(MODID, "preset_test");
+    public static final ResourceLocation NAMED_TEST = new ResourceLocation(MODID, "named_test");
+    public static final ResourceLocation NAMELESS_TEST = new ResourceLocation(MODID, "nameless_test");
+    
+    public MultiNoiseBiomeSourceLoadingEventTest()
+    {
+        IEventBus modBus = FMLJavaModLoadingContext.get().getModEventBus();
+        IEventBus forgeBus = MinecraftForge.EVENT_BUS;
+
+        modBus.addListener(this::onGatherData);
+        forgeBus.addListener(this::onLoadBiomeSource);
+        forgeBus.addListener(this::onRegisterCommands);
+    }
+    
+    private void onGatherData(GatherDataEvent event)
+    {
+        // need dynamic registries to do worldgen datagen
+        RegistryHolder registries = RegistryAccess.builtin();
+        RegistryWriteOps<JsonElement> writeOps = RegistryWriteOps.create(JsonOps.INSTANCE, registries);
+        
+        // the objects we reference must have been registered to the registryaccess
+        // creating a registry access deep-copies most of the BuiltinRegistries objects,
+        // so we must get vanilla objects from the registryaccess instead of getting static references
+        Registry<DimensionType> dimensionTypes = registries.registryOrThrow(Registry.DIMENSION_TYPE_REGISTRY);
+        Registry<NoiseGeneratorSettings> noiseSettings = registries.registryOrThrow(Registry.NOISE_GENERATOR_SETTINGS_REGISTRY);
+        Registry<NoiseParameters> noiseParameters = registries.registryOrThrow(Registry.NOISE_REGISTRY);
+        Registry<Biome> biomes = registries.registryOrThrow(Registry.BIOME_REGISTRY);
+        
+        DimensionType overworldDimensionType = dimensionTypes.get(DimensionType.OVERWORLD_LOCATION);
+        NoiseGeneratorSettings overworldNoiseSettings = noiseSettings.get(NoiseGeneratorSettings.OVERWORLD);
+        Biome desert = biomes.get(Biomes.DESERT);
+        
+        // make two three dimensions for testing
```

Comment:
Seems like a small mistake while writing:
```suggestion
        // make three dimensions for testing
```

Reply:
Fixed in next push